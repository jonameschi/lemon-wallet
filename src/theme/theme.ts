type Theme = {
  background: {
    inversed: string;
    primary: string;
    secondary: string;
    tertiary: string;
  };
  line: {
    primary: string;
  };
  skeletons: {
    background: string;
  };
  text: {
    inversed: string;
    negative: string;
    positive: string;
    primary: string;
    secondary: string;
    tertiary: string;
    warning: string;
  };
};

const theme: Theme = {
  background: {
    primary: '#F8F9FA',
    secondary: '#2647e4',
    tertiary: '#fff',
    inversed: '#D3D3D3',
  },
  line: {
    primary: '#DFE2E4',
  },
  text: {
    primary: '#212529',
    secondary: '#343A40',
    inversed: '#FFFFFF',
    tertiary: '#6C757D',
    negative: '#D90429',
    positive: '#21BF73',
    warning: '#f8b84e',
  },
  skeletons: {
    background: '#D3D3D3',
  },
};

type BackgroundColorKeys = keyof Theme['background'];
type TextColorKeys = keyof Theme['text'];

export { theme, Theme, BackgroundColorKeys, TextColorKeys };
