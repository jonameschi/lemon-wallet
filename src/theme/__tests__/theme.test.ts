import { theme } from 'theme/theme';

describe('Theme data', () => {
  it('should render a snapshot of Theme data', () => {
    expect(theme).toMatchSnapshot();
  });
});
