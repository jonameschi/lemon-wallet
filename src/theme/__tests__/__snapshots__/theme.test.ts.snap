// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme data should render a snapshot of Theme data 1`] = `
{
  "background": {
    "inversed": "#D3D3D3",
    "primary": "#F8F9FA",
    "secondary": "#2647e4",
    "tertiary": "#fff",
  },
  "line": {
    "primary": "#DFE2E4",
  },
  "skeletons": {
    "background": "#D3D3D3",
  },
  "text": {
    "inversed": "#FFFFFF",
    "negative": "#D90429",
    "positive": "#21BF73",
    "primary": "#212529",
    "secondary": "#343A40",
    "tertiary": "#6C757D",
    "warning": "#f8b84e",
  },
}
`;
